### PR TITLE
Set parallel_submit to True by default in job

### PR DESCRIPTION
**Pull request by mesmith75**
Seems worthwhile setting this to True by default.

Do we want to change the default in the `IBackend` as well?

### CODE REVIEW

**Comment by mesmith75** [comment]
retest this please

**Comment by mesmith75** [comment]
retest this please

**Comment by mesmith75** [comment]
retest this please

**Comment by mesmith75** [comment]
I think this works now - I had to redo a fair amount of the logic in `IBackend` and `Job`. I also overload the `master_submit` method in the Interactive backend to prevent it submitting in parallel ever.

File: python/Ganga/GPIDev/Adapters/IBackend.py [hunk 1]
```diff
@@ -72,23 +72,20 @@ def _parallel_submit(self, b, sj, sc, master_input_sandbox, fqid, logger):
...
             sj.updateStatus('submitting')
             if b.submit(sc, master_input_sandbox):
-                sj.updateStatus('submitted')
                 sj.info.increment()
+                return 1
             else:
                 raise IncompleteJobSubmissionError(fqid, 'submission failed')
         except Exception as err:
-            #from Ganga.Utility.logging import log_user_exception
-            sj.updateStatus('failed')
```
Comment (rob-c) [inline python/Ganga/GPIDev/Adapters/IBackend.py]: Shouldn't this update the master job to failed if some part of the whole set of jobs fails to submit?

File: python/Ganga/test/GPI/Bugs/TestSavannah17080.py [hunk 2]
```diff
@@ -12,12 +14,36 @@ def test_CondorConfigDefaults(self):
         j.backend= TestSubmitter()
         b = j.splitter.backs[1]
         b.fail = 'submit'
+        j.parallel_submit = False
 
         assert j.status == 'new'
 
-        j.submit(keep_going=False)
-        assert j.status in ['submitted', 'running']
+        with pytest.raises(Exception):
```
Comment (rob-c) [inline python/Ganga/test/GPI/Bugs/TestSavannah17080.py]: Can we be a little more specific with the type of exception raised here in a test which is known to fail?

Comment (mesmith75) [reply python/Ganga/test/GPI/Bugs/TestSavannah17080.py]: yes

File: python/Ganga/GPIDev/Adapters/IBackend.py (hunk 1 above)
Comment (mesmith75) [reply python/Ganga/GPIDev/Adapters/IBackend.py]: Yes but not here. That is taken care of in Job.py which catched the exception raised here.

**Comment by mesmith75** [comment]
retest this please

**Comment by mesmith75** [comment]
retest this please

**Pull request merged**